sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,538 @@
+
+Consideration: Technical
+
+Type: Standard
+
+Status: Draft
+
+Created: 10 October 2023
+
+License: BSD 2-Clause
+
+## Abstract
+
+This SIP proposes common RPC methods to use for the Stacks blockchain's "Connect" and "Auth" systems.
+The goal is to replace the current Connect interface, primarily used by web applications to connect with browser extensions and mobile apps with a more straightforward protocol.
+This proposal's goal is to standardize JSON compatible interfaces for use with wallet interfaces.
+
+## Introduction
+
+The current Connect system[^15], which has existed for several years, is primarily utilized by web applications for interfacing with wallets.
+However, many aspects of the existing "Connect" and "Auth" libraries are no longer required, leading to unnecessary complexity (e.g., wrapping RPC payloads in jsontokens) and lack of clear definitions (e.g., undefined serialization for non-JSON compatible data structures) in wallet connectivity.
+
+Recent attempts[^21][^22][^23][^24][^25][^26] to standardize the interface have sparked valuable discussions but have not culminated in a ratified standard, largely due to the stable state of the existing system.
+This SIP aims to address these issues by adopting the WBIPs standards[^11], which offer a more suitable RPC-style interface for modern web applications.
+The simplified protocol will allow integration without heavy dependencies (like Auth) and provide a more extendable interface for wallets.
+
+Additionally, this SIP is motivated by the increased traffic of Ordinal inscriptions on Bitcoin and the Stacks ecosystem growing closer to Bitcoin.
+The community has recognized the need for a more unified approach to wallet connectivity (e.g. Bitcoin and PSBTs for previously Stacks-only wallets).
+By adopting the new standard, we aim to align the community towards a common and modern protocol for wallet interaction in web applications.
+Importantly, the decision to use an existing standard (rather than designing a new one or reworking Auth) is intentional — to avoid further division or split ownership within the community.
+
+There was an attempt to re-use existing standards/protocols from other ecosystems via the WBIPs working group[^11][^26] — but no consensus was found that was a perfect fit or had enough traction for the larger layer-2 ecosystem.
+So this SIP aims to capture the important features for the Stacks ecosystem, with a focus on extensibility.
+
+## Specification
+
+The proposed changes are listed as follows:
+
+Specify [JSON-RPC 2.0](https://www.jsonrpc.org/specification) compatible methods and payloads for wallet interaction.
+These can be used via a browser object (i.e., via the `window.WalletProvider.request` method) or similar interfaces like WalletConnect.
+
+## Backwards Compatibility
+
+The implementation of this proposal is not necessarily backward compatible.
```

Comment:
Can we mention that most importantly, app private keys for domains have been removed.